Sources/Testing/Parameterization/Test.Case.ID.swift

```diff
@@ -8,7 +8,6 @@
 // See https://swift.org/CONTRIBUTORS.txt for Swift project authors
 //
 
-@_spi(ForToolsIntegrationOnly)
```

Comment:
I'm worried about this change -- making the `Identifiable` conformance public and also making its `ID` type public.  `Test.Case.ID` is not yet fully designed and, in some scenarios will not actually _uniquely_ identify a test case. When its `arguments` property is `nil` (which is the case if any argument is non-`Codable` or equivalent), it will be considered equal to all other instances representing non-`Codable` arguments. This could be confusing to users or lead to bugs.

This is a known limitation of this `ID` type currently which we need to come back and fix, but the current tool adopters already use it properly with that consideration in mind. But if we make this change it will publish this `ID` type as API and I don't think it is mature or robust enough yet.

The existing solution was to wrap this entire extension in `@_spi`. Can you elaborate on why that is problematic? A runtime `as? Identifiable` cast may succeed, but since the concrete `ID` type is not public, I did not think this would be a problem.

Reply:
Protocol conformances are always as visible as the type/protocol in question. See here: https://developer.apple.com/documentation/testing/test/case#relationships

If we don't want the conformance to be public, we'll have to remove it outright.

Reply:
Behold:

```swift
func idForTestCase(_ testCase: Test.Case) -> some Hashable {
  func thunk(_ x: some Identifiable) -> some Hashable {
    x.id
  }
  return thunk(testCase)
}
```

Contrived, but the point is the IDs are publicly accessible already.

Reply:
Actually, due to opaque types, this is valid even when everything's SPI and you haven't imported it:

```swift
let x = Test.Case.current!.id
let y = Test.Case.current!.id
#expect(x == y)
```

So yeah, we need to pick a side of the fence to be on here.

Reply:
I think it would certainly be preferable to remove the `Identifiable` conformance entirely, until the `ID` type is ready for broader use.

Also, the thing I was worried about was that this `ID` type might be used in `Equatable` comparisons. Your example showed a `Hashable` thunk, and luckily I think hash-ability is much less worrisome for this type. That said, we could also remove the `ID` type's conformance to `Equatable`, unless it's needed for some other reason I've forgotten. 

Either way, it sounds like we should completely remove `Test.Case`'s conformance to `Identifiable` for now.

Reply:
Yeah, let's just remove the Identifiable conformance

Reply:
@briancroom Thoughts?